packages/suite/src/components/suite/BioAuthGuard/BioAuthGuard.tsx

```diff
@@ -120,29 +91,34 @@ const BioAuthOverlay = ({
 };
 
 export const BioAuthGuard = ({ children }: { children: React.ReactNode }) => {
+    const [isWindowFocused, setIsWindowFocused] = useState(true);
     const isBioAuthValidationRequired = useSelector(state =>
         selectIsBioAuthValidationRequired(state, new Date()),
     );
-    const isBioAuthAvailable = useSelector(selectBioAuthEnabled);
-    const isAppUiHidden = false; // NOTE: temporary
-    const dispatch = useDispatch();
+    const isBioAuthEnabled = useSelector(selectBioAuthEnabled);
 
-    useEffect(() => {
-        dispatch(checkBioAuthAvailableThunk());
-    }, [dispatch]);
+    const { isBioAuthAvailable, requestBioAuthValidation } = useBioAuthDesktopApi();
```

Comment:
This used to be a simple component, with on clear `useEffect` that ran once to check if the bio auth is avaiable and one useEffect with listener that just checked if the window is blurred.

Now there are 2 comple use effects and one is hidden in the hook, is it really more mantainable?